eclipse-volttron/volttron-core
# fix for volttron non dev mode + poetry env

This depends on https://github.com/eclipse-volttron/volttron-core/pull/200
Code update to create a poetry project in VOLTTRON_HOME and update the current venv's packages to $VOLTTRON_HOME/pyproject.toml 
This is only applicable when volttron-zmq (or similar) is installed from pip once we push to pypi

Steps to test:
Since we don't have latest version of code in pypi, we have to manually test this with some code changes. I followed the following steps to test this

1. Create a new venv 
2. Cd into it, activate it
3. Create a new vhome and export it
4. pip install -e <point to source path of volttron-zmq>
5. pip list will show  volttron-core, volttron-lib-auth, and volttron-lib-zmq similar to packages from pypi (i.e. they won't be listed as src)
6. Change run_server.py line 146 - change "volttron==" to "volttron" 
7. start volttron without dev mode (volttron -vv -l volttron.log&)
8. verify that volttron starts without error
9. verify $VOLTTRON_HOME/pyproject.toml has all non volttron dependency
10. manually add volttron source libraries to pyproject.toml
11. poetry --directory <vhome> -e add <path>/volttron-core
12. poetry --directory <vhome> -e add <path>/volttron-lib-auth
13. poetry --directory <vhome> -e add <path>/volttron-lib-zmq
14. Now install volttron-listener (vctl install <path to listener>)
15. Verify that listener got added to pyproject.toml in vhome (poetry --directory <vhome> show volttron-listener)
